include/eve/module/complex/regular/detail/arithmetic.hpp

```diff
@@ -18,11 +18,17 @@ namespace eve::detail
   //==============================================================================================
   //  trivial extension of some real unary functions
   //==============================================================================================
-  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::ceil_,    Z const& z) noexcept { return Z{ceil(real(z)), ceil(imag(z))}; }
-  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::floor_,   Z const& z) noexcept { return Z{floor(real(z)), floor(imag(z))}; }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::ceil_,    Z const& z) noexcept { return Z{ceil(real(z)), ceil(imag(z))};       }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::floor_,   Z const& z) noexcept { return Z{floor(real(z)), floor(imag(z))};     }
   template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::nearest_, Z const& z) noexcept { return Z{nearest(real(z)), nearest(imag(z))}; }
-  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::trunc_,   Z const& z) noexcept { return Z{trunc(real(z)), trunc(imag(z))}; }
-  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::frac_,    Z const& z) noexcept { return Z{frac(real(z)), frac(imag(z))}; }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::trunc_,   Z const& z) noexcept { return Z{trunc(real(z)), trunc(imag(z))};     }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::frac_,    Z const& z) noexcept { return Z{frac(real(z)), frac(imag(z))};       }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::minus_,   Z const& z) noexcept { return Z{minus(real(z)), minus(imag(z))};     }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::plus_,    Z const& z) noexcept { return z;                                     }
+
+  template<decorator D, typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::minus_, D const &, Z const& z) noexcept { return minus(z);        }
```

Comment:
Why do we need those ?

Reply:
because minus is done for floating with a bitwise_xor

Reply:
but you are right I suppressed the plus and the decorated ones